pkg/sdk/data_types.go

```diff
@@ -112,3 +118,59 @@ func IsStringType(_type string) bool {
 		strings.HasPrefix(t, "NVARCHAR") ||
 		strings.HasPrefix(t, "NCHAR")
 }
+
+// ParseNumberDataTypeRaw extracts precision and scale from the raw number data type input.
+// It returns defaults if it can't parse arguments, data type is different, or no arguments were provided.
+// TODO [SNOW-1348103 or SNOW-1348106]: visit with functions and procedures rework
+func ParseNumberDataTypeRaw(rawDataType string) (int, int) {
+	r := util.TrimAllPrefixes(strings.TrimSpace(strings.ToUpper(rawDataType)), DataTypeNumberSynonyms...)
```

Comment:
I don't get why this part is needed here `strings.TrimSpace(strings.ToUpper(rawDataType))` and that it works 🤔, because having string `NUMBER(30)` and trim prefix `NUMBER(30)` we should end up with `r` equal to an empty string or am I missing something.

Reply:
We are trimming a prefix without parentheses, so e.g. `number(30)` becomes `(30)` and we process the remaining part to extract attribute values.
Trimming spaces was added to avoid stupid whitespace problems e.g. ` number ( 30 ) `.